tools/caffe_converter/convert_model.py

```diff
@@ -46,25 +46,25 @@ def main():
         layers = net_caffe.layers
     else:
         layers = parse.parse_caffemodel(args.caffe_model)
-    
+
     arg_shapes, output_shapes, aux_shapes = prob.infer_shape(data=tuple(input_dim))
     arg_names = prob.list_arguments()
     arg_shape_dic = dict(zip(arg_names, arg_shapes))
     arg_params = {}
-    
+
     iter = ''
     if caffe_flag:
         iter = get_caffe_iter(layer_names, layers)
     else:
         iter = get_iter(layers)
     first_conv = True
-    
+
     for layer_name, layer_type, layer_blobs in iter:
         if layer_type == 'Convolution' or layer_type == 'InnerProduct' or layer_type == 4 or layer_type == 14:
             assert(len(layer_blobs) == 2)
             wmat_dim = []
-            if len(layer_blobs[0].shape.dim) > 0:
-                wmat_dim = layer_blobs[0].shape.dim
+            if len(layer_blobs[0].shape) > 0:
+                wmat_dim = [layer_blobs[0].shape[i] for i in xrange(len(layer_blobs[0].shape))]
```

Comment:
Caffe changed from blobs with a fixed 4 dimensions to blobs of arbitrary dimension. This probably caused the issue. But that change has been around for quite some time.